addons/core/addon/components/dropdown/index.js

```diff
@@ -70,6 +70,16 @@ export default class DropdownComponent extends Component {
     onClose();
   }
 
+  /**
+   * Sets selectedItems to checkedItems when the user opens
+   * the dropdown and potentially has made changes
+   * outside of the scope of dropdown
```

Comment:
Out of curiosity when does it happen where changes were made outside the dropdown where this use case gets hit?

Reply:
When we dismiss a tag or clear all filters. We are updating the route queryParams but the `selectedItems` inside the dropdown does not get updated. It still has value. So when we open the dropdown, it will check the `@checkedItems` param to verify if any state has changed outside the scope of the dropdown.